mesa/visualization/components/matplotlib.py

```diff
@@ -327,11 +329,14 @@ def PlotMatplotlib(model, measure, dependencies: list[any] | None = None):
     elif isinstance(measure, dict):
         for m, color in measure.items():
             ax.plot(df.loc[:, m], label=m, color=color)
-        fig.legend()
+        ax.legend(loc="best")
     elif isinstance(measure, list | tuple):
         for m in measure:
             ax.plot(df.loc[:, m], label=m)
-        fig.legend()
+        ax.legend(loc="best")
```

Comment:
I'm not sure if "best" is already the default when not specified, but a problem is that the legend location might change over time depending on the lines being plotted.

Reply:
Yes it does change over time depending on how lines are plotted. If we want the legend to be at a fixed location, I tried to make it at top right corner outside the main axis. Then the legend will never intersect with any plot. However, the issue is that the plotting area becomes smaller.

Reply:
I think it's fine if it moves, that just means it's more out of the way again.